ext/ptrace/ptrace.c

```diff
@@ -21,7 +23,18 @@ ptrace_peekdata(VALUE _mod, VALUE pid, VALUE addr, VALUE _data) {
 
 static VALUE
 ptrace_peekuser(VALUE _mod, VALUE pid, VALUE off, VALUE _data) {
-  long val = ptrace(PTRACE_PEEKUSER, NUM2LONG(pid), NUM2LONG(off), NULL);
+  // aarch64 doesn't support PTRACE_PEEKUSER, but it does support
+  // PTRACE_GETREGSET which is fairly similar. Since i386 and x86_64 support it
+  // too, just use that unconditionally to present the same API for the
+  // registers we care about.
+  size_t offset = NUM2LONG(off);
```

Comment:
Can we check  this value is valid? Something like it should be less than `sizeof(struct user_pt_regs)`

Reply:
Ideally yes, but see above.